### PR TITLE
OHRM5X-1205: Access employee lang string

### PR DESCRIPTION
**Pull request by Ajiharan007**

### CODE REVIEW

File: symfony/client/src/orangehrmPimPlugin/components/FullNameInput.vue [hunk 1]
```diff
@@ -76,6 +77,12 @@ export default {
       type: Boolean,
       default: false,
     },
+    middleNameProps: {
+      type: Object,
+      default: () => ({
+        placeholder: 'Middle Name',
```
Comment (RajithaKumara) [inline symfony/client/src/orangehrmPimPlugin/components/FullNameInput.vue:83]: have to check i18n support

**Comment by RajithaKumara** [comment]
Have to cherry-pick from [here](https://github.com/orangehrm/orangehrm/pull/976/commits/0e3f3c8a6b7509c621f4a2a5e4d3da47a1fcd8e6)
```bash
git fetch upstream
git checkout -b OHRM5X-1205--Access-employee-lang-string upstream/5.0-i18n
git cherry-pick 3a1392c^..3771962
# resolve conflicts
```

File: symfony/client/src/orangehrmPimPlugin/components/FullNameInput.vue [hunk 2]
```diff
@@ -78,10 +77,8 @@ export default {
       default: false,
     },
     middleNameProps: {
-      type: Object,
-      default: () => ({
-        placeholder: "$t('middle_name')",
```
Comment (Ajiharan007) [inline symfony/client/src/orangehrmPimPlugin/components/FullNameInput.vue]: cannot use this keyword in default value

File: symfony/client/src/orangehrmPimPlugin/components/FullNameInput.vue [hunk 3]
```diff
@@ -78,10 +77,8 @@ export default {
       default: false,
     },
     middleNameProps: {
-      type: Object,
-      default: () => ({
-        placeholder: 'Middle Name',
-      }),
+      type: Boolean,
+      default: false,
     },
```
Comment (RajithaKumara) [inline symfony/client/src/orangehrmPimPlugin/components/FullNameInput.vue:82]: shall we go like this then,
```js
showMiddleNamePlaceholder: {
      type: Boolean,
      default: true,
}

File: symfony/client/src/orangehrmPimPlugin/components/FullNameInput.vue [hunk 4]
```diff
@@ -33,11 +33,10 @@
     <oxd-input-field
       class="orangehrm-middlename"
       name="Middle Name"
-      :placeholder="$t('general.middle_name')"
       :model-value="middleName"
       :rules="rules.middleName"
       :disabled="disabled"
-      v-bind="middleNameProps"
+      :placeholder="middleNameProps ? middleName : $t('general.middle_name')"
```
Comment (RajithaKumara) [inline symfony/client/src/orangehrmPimPlugin/components/FullNameInput.vue:39]:
```suggestion
      :placeholder="showMiddleNamePlaceholder ? $t('general.middle_name') : ''"
```

File: symfony/client/src/orangehrmMaintenancePlugin/components/SelectedEmployee.vue [hunk 5]
```diff
@@ -44,7 +44,7 @@
                   :middle-name="selectedEmployee.middleName"
                   :last-name="selectedEmployee.lastName"
                   :rules="rules"
-                  :middle-name-props="{}"
+                  middle-name-props
```
Comment (RajithaKumara) [inline symfony/client/src/orangehrmMaintenancePlugin/components/SelectedEmployee.vue:47]:
```suggestion
                  :show-middle-name-placeholder="false"
```

**Review by RajithaKumara — CHANGES_REQUESTED** [review]
Shall we do it like this?

**Pull request merged**